client/src/main/java/com/paypal/selion/platform/html/AbstractElement.java

```diff
@@ -485,6 +485,9 @@ public void click(Object... expected) {
         if (expected == null || expected.length == 0) {
             return;
         }
+        if (parent != null) {
+            WebDriverWaitUtils.waitUntilPageIsLoaded(parent.getCurrentPage());
+        }
```

Comment:
This is a very good question, I think you're right! In this case, loading check might be excessive. Should I move `waitUntilPageIsLoaded` inside the for loop and apply to first three if statements ?